website/static/js/tests/addons/folderPickerNodeConfig.test.js

```diff
@@ -0,0 +1,450 @@
+            });
+        });
+        describe('#showTokenCreateButton', () => {
+            var reset = () => {
+                vm.userHasAuth(false);
+                vm.nodeHasAuth(false);
+                vm.loadedSettings(true);
+            };
+            it('shows the token create button if the User doesn\'t have auth, the Node doesn\'t have auth, and the VM has loaded settings', () => {
+                reset();
+                assert.isTrue(vm.showTokenCreateButton());
+            });
+            it('... and doesn\'t show the token create button otherwise', () => {
+                reset();
+                vm.userHasAuth(true);
+                assert.isFalse(vm.showTokenCreateButton());
+                reset();
+                vm.nodeHasAuth(true);
+                assert.isFalse(vm.showTokenCreateButton());
+                reset();
+                vm.loadedSettings(false);
+                assert.isFalse(vm.showTokenCreateButton());
+                reset();
+                vm.userHasAuth(true);
+                vm.nodeHasAuth(true);
+                vm.loadedSettings(false);
+                assert.isFalse(vm.showTokenCreateButton());
+            });
+        });
+        describe('#folderName', () => {
+            it('returns the value of the name property of the currently set folder when the Node has auth', () => {
+                vm.nodeHasAuth(true);
+                vm.folder({
+                    name: null,
+                    id: null
+                });
+                assert.isNull(vm.folderName());
+                var name = faker.hacker.noun();
+                vm.folder({
+                    name: name,
+                    id: faker.finance.account()
+                });
+                assert.equal(vm.folderName(), name);
+            });
+            it('... and returns an empty string otherwise', () => {
+                vm.nodeHasAuth(false);
+                assert.equal(vm.folderName(), '');
+            });
+        });
+        describe('#selectedFolderName', () => {
+            it('returns the selected folder\'s name if set else \'None\' when the User is owner', () => {
+                vm.userIsOwner(true);
+                vm.selected({
+                    name: null,
+                    id: null
+                });
+                assert.equal(vm.selectedFolderName(), 'None');
+                var name = faker.hacker.noun();
+                assert.notEqual(name, 'None');
+                vm.selected({
+                    name: name,
+                    id: faker.finance.account()
+                });
+                assert.equal(vm.selectedFolderName(), name);
+            });
+            it('... and returns an empty string otherwise', () => {
+                vm.userIsOwner(false);
+                assert.equal(vm.selectedFolderName(), '');
+            });
+        });
+        describe('#changeMessage', () => {
+            var reset = () => {
```

Comment:
It could be used in changeMessage here: https://github.com/samchrisinger/osf.io/blob/435a1522a33dfd99de2cb1aa2afb7015631f4af8/website/static/js/folderPickerNodeConfig.js#L176-177